courses/views/v1/__init__.py

```diff
@@ -190,3 +204,38 @@ def destroy(self, request, *args, **kwargs):
         if deactivated_enrollment is None:
             return Response(status=status.HTTP_400_BAD_REQUEST)
         return Response(status=status.HTTP_204_NO_CONTENT)
+
+    def partial_update(self, request, *args, **kwargs):
+        enrollment = self.get_object()
+        receive_emails = request.data.get("receive_emails", None)
+
+        if receive_emails is not None:
+            # means if receive_emails is passed in the request body
+            with transaction.atomic():
+                try:
+                    if receive_emails:
+                        response = subscribe_to_edx_course_emails(
+                            request.user, enrollment.run
+                        )
+                        enrollment.edx_emails_subscription = True
+                    else:
+                        response = unsubscribe_from_edx_course_emails(
+                            request.user, enrollment.run
+                        )
+                        enrollment.edx_emails_subscription = False
+                    enrollment.save()
+                    return Response(data=response, status=status.HTTP_200_OK)
+                except (
+                    EdxApiEmailSettingsErrorException,
+                    UnknownEdxApiEmailSettingsException,
+                    NoEdxApiAuthError,
+                    HTTPError,
+                    RequestsConnectionError,
+                ) as exc:
+                    log.exception(str(exc))
+                    return Response(data=str(exc), status=status.HTTP_400_BAD_REQUEST)
+        else:
+            # only designed to update edx_emails_subscription field
+            # TODO: if needed ever in future please add implementation 
+            # to update rest of the fields in PATCH request
```

Comment:
TODOs are not usually "if needed" but something that makes sense to update when possible.
```suggestion
            # Only designed to update edx_emails_subscription field
            # TODO: In the future please add the implementation 
            # to update the rest of the fields in the PATCH request
            # or separate out the APIs into function-based views.
```